posthog/hogql/test/_test_parse_string.py

```diff
@@ -0,0 +1,61 @@
+from typing import Literal
+from posthog.hogql.errors import SyntaxException
+from posthog.hogql.parse_string import parse_string as parse_string_py
+from hogql_parser import unquote_string as unquote_string_cpp
+from posthog.test.base import BaseTest
+
+
+def parse_string_test_factory(backend: Literal["python", "cpp"]):
+    parse_string = parse_string_py if backend == "python" else unquote_string_cpp
+
+    class TestParseString(BaseTest):
+        def test_quote_types(self):
+            self.assertEqual(parse_string("`asd`"), "asd")
+            self.assertEqual(parse_string("'asd'"), "asd")
+            self.assertEqual(parse_string('"asd"'), "asd")
+            self.assertEqual(parse_string("{asd}"), "asd")
+
+        def test_escaped_quotes(self):
+            self.assertEqual(parse_string("`a``sd`"), "a`sd")
+            self.assertEqual(parse_string("'a''sd'"), "a'sd")
+            self.assertEqual(parse_string('"a""sd"'), 'a"sd')
+            self.assertEqual(parse_string("{a{{sd}"), "a{sd")
+            self.assertEqual(parse_string("{a}sd}"), "a}sd")
+
+        def test_escaped_quotes_slash(self):
+            self.assertEqual(parse_string("`a\\`sd`"), "a`sd")
+            self.assertEqual(parse_string("'a\\'sd'"), "a'sd")
+            self.assertEqual(parse_string('"a\\"sd"'), 'a"sd')
+            self.assertEqual(parse_string("{a\\{sd}"), "a{sd")
+
+        def test_slash_escape(self):
+            self.assertEqual(parse_string("`a\nsd`"), "a\nsd")
+            self.assertEqual(parse_string("`a\\bsd`"), "a\bsd")
+            self.assertEqual(parse_string("`a\\fsd`"), "a\fsd")
+            self.assertEqual(parse_string("`a\\rsd`"), "a\rsd")
+            self.assertEqual(parse_string("`a\\nsd`"), "a\nsd")
+            self.assertEqual(parse_string("`a\\tsd`"), "a\tsd")
+            self.assertEqual(parse_string("`a\\asd`"), "a\asd")
+            self.assertEqual(parse_string("`a\\vsd`"), "a\vsd")
+            self.assertEqual(parse_string("`a\\\\sd`"), "a\\sd")
+            self.assertEqual(parse_string("`a\\0sd`"), "asd")
```

Comment:
This is the only change in this whole test suite: NUL is no longer supported, because Python ignores it in `PyUnicode_FromStringAndSize`. This is surprising to me, but I really couldn't figure out a way to keep the NUL in the str constructed from C++